clkmgr/README.md

```diff
@@ -0,0 +1,114 @@
+synchronization on the local platform. The Clock Manager exposes an API to
+enable reporting time synchronization status to any application executing on the
+system. It is planned that Clock Manager will not only be able to report
+timing telemetry information, but also configure the platform, allowing
+privileged applications to make changes to the platform time configuration.
+
+![High-level Overview of the Clock Manager
+Functionality](./image/clkmgr_overview.png) __Figure 1 - High-level Overview of
+the Clock Manager Functionality__
+
+Figure 1 shows the high-level system interactions between the Clock Manager and
+the system software on the local platform. The Clock Manager communicates with
+the time synchronization daemon (e.g., Linux PTP or Chrony) to get current
+telemetry data and synchronization error and notifies the application of changes
+to the relevant state. Any changes to the relevant state will generate event
+notifications to the application. This is shown by Clock Manager data path. The
+Clock Manager data path is entirely within user-space except for the IPC and
+does not introduce any additional kernel modules. The current Linux timekeeping
+data path is unchanged.
+
+# Architectural Considerations
+
+For more details about the Clock Manager design, supported events, and current
+APIs, please check out the [Design Document](HLD_clkmgr.md). To check out how to
+see the Clock Manager in action, please visit the [demo
+application](TEST_clkmgr.md). 
+
+## Maintaining Expected Timekeeping Functionality
+
+The Linux timekeeping data path is the fast path. Typically, reading the current
+time using vDSO calls such as clock_gettime() incurs only a few tens of
+nanoseconds and no context switch. The complete timekeeping data path from time
+sync daemon to the application – including clock_adjtime() – incurs a typical
+latency of less than 10 microseconds. The relatively slower and parallel Clock
+Manager data path requires additional processing and incurs up to 10
+milliseconds of latency. This design ensures that the performance of the
+existing timekeeping data path in the kernel is unchanged by the Clock Manager
+software.
+
+## Licensing 
+
+Applications utilizing Clock Manager must be allowed to be licensed and released
+under any free or non-free terms. The time synchronization daemon (e.g., Linux
+PTP or Chrony) are licensed under GPL. The Clock Manager framework insulates the
+application from any requirement to release the source code. The Clock Manager
+service is implemented using a process separate from both the application and
+time synchronization daemon. This process is the Clock Manager Proxy (CMP). The
+CMP code will include GPL and/or link to GPL code making it a derivative and
+requiring that all code in the CMP code be licensed under the GPLv2. Any Clock
+Manager support libraries must be released under a GPL-compatible license that
+can also be used with proprietary software, such as MIT or modified BSD
```

Comment:
We decided to use BSD-3-Clause license, so we can explicitly mention the license.

Reply:
Updated